### PR TITLE
Add test suite

### PR DESCRIPTION
**Pull request by blink1073**
```
SUMMARY:
✔ 31 tests completed
------------|----------|----------|----------|----------|----------------|
File        |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
------------|----------|----------|----------|----------|----------------|
 lib/       |    98.13 |    84.78 |      100 |      100 |                |
  index.js  |      100 |       50 |      100 |      100 |                |
  layout.js |    97.18 |    89.19 |      100 |      100 |                |
  panel.js  |      100 |    71.43 |      100 |      100 |                |
------------|----------|----------|----------|----------|----------------|
All files   |    98.13 |    84.78 |      100 |      100 |                |
------------|----------|----------|----------|----------|----------------|
```

### CODE REVIEW

File: src/layout.ts [hunk 1]
```diff
@@ -139,6 +139,11 @@ class PanelLayout extends AbstractLayout {
    */
   protected attachChild(index: number, child: Widget): void {
     let ref = this.childAt(index + 1);
+    if (ref && ref.node) {
+      if (!this.parent.node.contains(ref.node)) {
+        ref = null;
+      }
+    }
```
Comment (blink1073) [inline src/layout.ts]: Avoids this error:

```
Error: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
```

File: test/src/layout.ts [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        expect(layout.methods.indexOf('initialize')).to.not.be(-1);
+      });
+
+      it('should attach widget children', () => {
+        let widget = new Widget();
+        let children = [new LogWidget(), new LogWidget()];
+        let layout = new LogLayout();
+        layout.addChild(children[0]);
+        layout.addChild(children[1]);
+        widget.attach(document.body);
```
Comment (blink1073) [inline test/src/layout.ts]: This is the test that triggered the error above.

File: src/layout.ts (hunk 1 above)
Comment (sccolbert) [inline src/layout.ts]: This should not be needed. Please remove and I'll check in the failing test and fix it.

**Comment by blink1073** [comment]
Reverted.

**Pull request merged**